### PR TITLE
KE-31356 kyspark-3.2.x -4.x fix ut

### PR DESCRIPTION
**Pull request by leejaywei**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)
(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

Please review http://spark.apache.org/contributing.html before opening a pull request.

### CODE REVIEW

**Comment by leejaywei** [comment]
retest this please

**Comment by kyligence-git** [comment]
Can one of the admins verify this patch?

**Comment by leejaywei** [comment]
retest this please

**Comment by leejaywei** [comment]
retest this please

**Review by zheniantoushipashi — DISMISSED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 1]
```diff
@@ -855,7 +861,7 @@ class AnalysisSuite extends AnalysisTest with Matchers {
           UnresolvedFunction(f, $"a" :: Nil, isDistinct = false))
         testAnalyzer1.execute(plan1)
         // One-parameter is not deprecated.
-        assert(logAppender.loggingEvents.isEmpty)
+        check(0)
```
Comment (yugan91sh) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:864]: check(0)是在干啥？

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala [hunk 2]
```diff
@@ -209,15 +210,15 @@ class DateTimeUtilsSuite extends SparkFunSuite with Matchers with SQLHelper {
       checkStringToTimestamp("2015-03-18T12:03:17-01:00", expected)
       checkStringToTimestamp("2015-03-18T12:03:17GMT-01:00", expected)
 
-      zoneId = getZoneId("07:30")
+      zoneId = getZoneId("+07:30")
```
Comment (yugan91sh) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala:213]: 时区格式会受到什么的影响？

**Review by yugan91sh — APPROVED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala (hunk 2 above)
Comment (leejaywei) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/util/DateTimeUtilsSuite.scala:213]: image
image
不带+，- 会解析出错

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala (hunk 1 above)
Comment (leejaywei) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:864]: 这里表示符合条件的警告日志出现了0次

**Pull request merged**